zebrad/src/commands/start/sync.rs

```diff
@@ -82,62 +91,51 @@ where
         let mut download_set = HashSet::new();
         while let Some(res) = tip_futs.next().await {
             match res.map_err::<Report, _>(|e| eyre!(e)) {
-                Ok(zebra_network::Response::BlockHeaderHashes(hashes)) => {
-                    info!(
-                        new_hashes = hashes.len(),
-                        in_flight = self.block_requests.len(),
-                        downloaded = self.downloaded.len(),
-                        "requested more hashes"
-                    );
-
-                    // TODO(jlusby): reject both main and test net genesis blocks
-                    if hashes.last() == Some(&super::GENESIS) {
+                Ok(zn::Response::BlockHeaderHashes(hashes)) => {
+                    if hashes.is_empty() {
+                        tracing::debug!("skipping empty response");
                         continue;
                     }
 
-                    let mut hashes = hashes.into_iter().peekable();
```

Comment:
instead of dealing with peekable iterators if you're going to grab the tip from the back I'd just use `pop()` on `Vec` to immediately separate the tip from the rest of the blocks

Reply:
I don't think we should separate the tip from the rest of the blocks, because it means we have to separate the tip download logic separately from the logic for downloading the rest of the blocks.

Reply:
we could just use the prospective_tips to extend the download set right before we pass the download_set to `request_blocks`